masq/src/commands/financials_command.rs

```diff
@@ -787,37 +816,59 @@ impl FinancialsCommand {
 
     fn process_optionally_fractional_number<N>(num: &str) -> Result<N, String>
     where
-        N: FromStr<Err = ParseIntError> + TryFrom<i64> + CheckedMul + Display,
+        N: FromStr<Err = ParseIntError> + TryFrom<i64> + TryFrom<u64> + CheckedMul + Display + Copy,
+        i64: TryFrom<N>,
+        u64: TryFrom<N>,
         <N as TryFrom<i64>>::Error: Debug,
+        <i64 as TryFrom<N>>::Error: Debug,
     {
-        fn decimal_digits_count(num: &str, dot_idx: usize) -> usize {
-            num.chars().count() - dot_idx - 1
+        const DIGITS_IN_BILLION: u32 = 9;
+        fn all_digits_with_dot_removed(num: &str) -> String {
+            num.chars().filter(|char| *char != '.').collect()
         }
-
-        let final_unit_conversion = |parse_result: N, pow_factor: u32| {
+        fn check_if_all_decimal_digits_fits_in_gwei_range(
+            digits_count_unchecked: u32,
+            num: &str,
+        ) -> Result<u32, String> {
+            if digits_count_unchecked <= DIGITS_IN_BILLION {
+                Ok(digits_count_unchecked)
+            } else {
+                Err(format!("Value '{}' exceeds the limit of maximally nine decimal digits (only gwei supported)", num))
+            }
+        }
+        fn number_of_decimal_digits(num: &str, dot_idx: usize) -> u32 {
+            u32::try_from(num.chars().count() - dot_idx - 1)
```

Comment:
Maybe a short comment at the end of this line? ("// Dot was removed")
Maybe also rename `dot_idx` to `former_dot_idx` or something?